fastly/director_test.go

```diff
@@ -86,17 +129,20 @@ func TestClient_Directors(t *testing.T) {
 	if err != nil {
 		t.Fatal(err)
 	}
-	if b.Name != nb.Name {
-		t.Errorf("bad name: %q (%q)", b.Name, nb.Name)
+	if d.Name != nb.Name {
+		t.Errorf("bad name: %q (%q)", d.Name, nb.Name)
+	}
+	if d.Quorum != nb.Quorum {
+		t.Errorf("bad quorum: %q (%q)", d.Quorum, nb.Quorum)
 	}
-	if b.Quorum != nb.Quorum {
-		t.Errorf("bad quorum: %q (%q)", b.Quorum, nb.Quorum)
+	if d.Type != nb.Type {
+		t.Errorf("bad type: %q (%q)", d.Type, nb.Type)
 	}
-	if b.Type != nb.Type {
-		t.Errorf("bad type: %q (%q)", b.Type, nb.Type)
+	if d.Retries != nb.Retries {
+		t.Errorf("bad retries: %q (%q)", d.Retries, nb.Retries)
 	}
-	if b.Retries != nb.Retries {
-		t.Errorf("bad retries: %q (%q)", b.Retries, nb.Retries)
+	if len(nb.Backends) == 0 || nb.Backends[0] != b.Name {
+		t.Error("bad backend: expected a backend")
 	}
```

Comment:
This test assertion validates that the requested Director contains the Backend we expect it to.